File: 141. 环形链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if(head==NULL)
            return false;
        ListNode *l1=head,*l2=head->next;
        while(l1!=NULL&&l2!=NULL&&l2->next!=NULL)
        {
            if(l1->val==l2->val)
                return true;
            l1=l1->next;
            l2=l2->next->next;
        }
        return false;
    }
};

File: js/141. 环形链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let cache=[]
    let current=head
    while(current){
        if(cache.indexOf(current)>=0)
            return true
        cache.push(current)    
        current=current.next
    }
    return false
};
